src/projectile.cpp

```diff
@@ -91,153 +92,39 @@ void projectile::unset_custom_explosion()
 
 void apply_ammo_effects( const tripoint &p, const std::set<std::string> &effects )
 {
-    if( effects.count( "EXPLOSIVE_RAUFOSS" ) > 0 ) {
-        // TODO: Make this actually fill a ~4m radius circle with fire and 20 shrapnel bits.  Or better, a 30 deg, 15m long arc.
-        explosion_handler::explosion( p, 2.4, 0.6, true, 28, 1.4 );
-    }
-
-    if( effects.count( "EXPLOSIVE_SMALL" ) > 0 ) {
-        // TODO: double-check if this is sensible.
-        explosion_handler::explosion( p, 360, 0.4 );
-    }
-
-    if( effects.count( "EXPLOSIVE" ) > 0 ) {
-        // TODO: double-check if this is sensible.
-        explosion_handler::explosion( p, 360 );
-    }
-
-    if( effects.count( "FRAG" ) > 0 ) {
-        // Same as a standard thrown frag grenade.
-        explosion_handler::explosion( p, 185, 0.8, false, 212, 0.05 );
-    }
-
-    if( effects.count( "NAPALM" ) > 0 ) {
-        explosion_handler::explosion( p, 60, 0.7, true );
-        // More intense fire near the center
-        for( auto &pt : g->m.points_in_radius( p, 1, 0 ) ) {
-            g->m.add_field( pt, fd_fire, 1 );
-        }
-    }
-
-    if( effects.count( "NAPALM_BIG" ) > 0 ) {
-        explosion_handler::explosion( p, 360, 0.8, true );
-        // More intense fire near the center
-        for( auto &pt : g->m.points_in_radius( p, 3, 0 ) ) {
-            g->m.add_field( pt, fd_fire, 1 );
-        }
-    }
-
-    if( effects.count( "PYROPHORIC" ) > 0 ) {
-        explosion_handler::explosion( p, 360, 0.8, true );
-        // Extreme heat near the center of the explosion
-        for( auto &pt : g->m.points_in_radius( p, 3, 0 ) ) {
-            g->m.add_field( pt, fd_fire, 2 );
-        }
-    }
-
-    if( effects.count( "MININUKE_MOD" ) > 0 ) {
-        explosion_handler::explosion( p, 72000000 );
-        for( auto &pt : g->m.points_in_radius( p, 18, 0 ) ) {
-            if( g->m.sees( p, pt, 3 ) &&
-                g->m.passable( pt ) ) {
-                g->m.add_field( pt, fd_nuke_gas, 3 );
+    for( const ammo_effect &ae : ammo_effects::get_all() ) {
+        if( effects.count( ae.id.str() ) > 0 ) {
+            for( auto &pt : g->m.points_in_radius( p, ae.aoe_radius, ae.aoe_radius_z ) ) {
+                if( one_in( ae.aoe_chance ) ) {
+                    const bool check_sees = !ae.aoe_check_sees ||
+                                            ( ae.aoe_check_sees && g->m.sees( p, pt, ae.aoe_check_sees_radius ) );
+                    const bool check_passable = !ae.aoe_check_passable ||
```

Comment:
This can be reduce:
```suggestion
                    const bool check_passable = !ae.aoe_check_passable || g->m.passable( pt );
```
The second check for `ae.aoe_check_passable` is not required. `g->m.passable( pt )` will only be checked when `ae.aoe_check_passable` is true.
Same for the above check.